compiler/rustc_hir/src/hir.rs

```diff
@@ -841,7 +843,28 @@ impl<'hir> Generics<'hir> {
 
 /// A single predicate in a where-clause.
 #[derive(Debug, Clone, Copy, HashStable_Generic)]
-pub enum WherePredicate<'hir> {
+pub struct WherePredicate<'hir> {
+    pub hir_id: HirId,
+    pub kind: &'hir WherePredicateKind<'hir>,
+}
```

Comment:
Same remark as the AST: should we take the opportunity to move the `Span` here? And eventually remove the `HirId` from `WhereBoundPredicate`?

Reply:
Cool! That's much clearer.